app/src/main/java/comp3350/mealbuddy/business/GoalFactory/GoalFactory.java

```diff
@@ -0,0 +1,85 @@
+package comp3350.mealbuddy.business.GoalFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import comp3350.mealbuddy.objects.CalorieGoal;
+import comp3350.mealbuddy.objects.Goal;
+import comp3350.mealbuddy.objects.UserInfo;
+
+import static comp3350.mealbuddy.objects.UserInfo.Sex.MALE;
+
+
+public abstract class GoalFactory {
+
+    protected UserInfo userInfo;
+
+    public GoalFactory(UserInfo userInfo) {
+        this.userInfo = userInfo;
+    }
+
+    public List<Goal> makeGoals(){
+        List<Goal> goalList = new ArrayList<>();
+        addTotalCalorieGoal(userInfo, goalList);
+        addMacroGoals(userInfo, goalList);
+        addMicroGoals(userInfo, goalList);
+        addLabelGoals(userInfo, goalList);
+        return goalList;
+    }
+
+    protected abstract void addMacroGoals(UserInfo userInfo, List<Goal> goalList);
+    protected abstract void addMicroGoals(UserInfo userInfo, List<Goal> goalList);
+    protected abstract void addLabelGoals(UserInfo userInfo, List<Goal> goalList);
+
+    private void addTotalCalorieGoal(UserInfo userInfo, List<Goal> goalList){
+        int recCalAmount;
+        int calVariance = 200;
+        if(userInfo.sex == MALE){
+            recCalAmount = getMaleTotalCalorieRecAmnt(userInfo);
+        }else{
+            recCalAmount = getFemaleTotalCalorieRecAmnt(userInfo);
+        }
```

Comment:
```suggestion
        int recCalAmount = (userInfo.sex == MALE) 
            ?   getMaleTotalCalorieRecAmnt(userInfo)
            :    getFemaleTotalCalorieRecAmnt(userInfo)
        int calVariance = 200;
```

One option to make this cleaner is to a ternary operator.

Second option, is to make a function like this getTotalCalorieRecAmount(userInfo) which checks the sex, and calls the female or male classes accordingly. the benefit of this, is you can just call this function directly everytime and don't have to check the sex.